point_pattern.py

```diff
@@ -119,7 +138,19 @@ def average_nearest_neighbor_distance(points):
      Measure of Spatial Relationships in Populations. Ecology. 35(4)
      p. 445-453.
     """
-    mean_d = 0
+    nearest = []
+
+    for n, point in enumerate(points):
+        nearest.append(None)
+        for point2 in points:
+            if point is not point2:
```

Comment:
Check out `enumerate`.  This works when two valid points are not coincident.  if they are, you could use positional index in the list to see if they are the same.